v2/googlecloud-to-splunk/docs/GCSToSplunk/README.md

```diff
@@ -0,0 +1,109 @@
+# GCS To Splunk Dataflow Template
+
+The [GCSToSplunk](../../src/main/java/com/google/cloud/teleport/v2/templates/GCSToSplunk.java) pipeline ingests
+data from CSV files stored in GCS, optionally executes a UDF,
+and writes those records into Splunk's HEC endpoint.
+
+The template creates the Splunk payload as a JSON element using one of the following:
+
+1. CSV headers* (default option, used only if none of the below two inputs were provided)
+2. JSON schema (if provided)
+3. Javascript UDF (if provided)
+
+
+If a Javascript UDF and JSON schema are both inputted as parameters, 
+only the Javascript UDF will be executed.
+
+
+## Getting Started
+
+### Requirements
+
+* Java 8
+* Maven
+* Cloud Storage bucket exists
+* Splunk HEC instance exists
+
+### Building Template
+
+This is a Flex Template meaning that the pipeline code will be containerized, and the container will be
+run on Dataflow.
+
+#### Building Container Image
+
+__Set environment variables that will be used in the build process.__
```

Comment:
Section headers are identified by `#` symbols. It would look something like:

```
#### Set Environment Variables

...

#### Set Cloud Project

...

#### Build and Push Image
```

Also, I don't think setting the Cloud project is necessary, since no `gcloud` commands are run under `Building Template`, so you can leave that out. Whether or not to combine the other two sections is up to you.